middleware.go

```diff
@@ -20,6 +20,28 @@ func (mw MiddlewareFunc) Middleware(handler http.Handler) http.Handler {
 	return mw(handler)
 }
 
+// NewMiddleware helper method which allows creating a middleware from a function which receives the response and request.
+// The typical case would be passing a closure
+func NewMiddleware(mw func(http.ResponseWriter, *http.Request)) MiddlewareFunc {
+	return func(handler http.Handler) http.Handler {
+		return &genericMW{
+			process: func(w http.ResponseWriter, r *http.Request) {
+				mw(w, r)
+				handler.ServeHTTP(w, r)
+			},
+		}
+	}
+}
+
+// genericMW is for creating a middleware from a closure, it's used in the public method NewMiddleware
+type genericMW struct {
+	process func(http.ResponseWriter, *http.Request)
+}
+
+func (instance *genericMW) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	instance.process(w, req)
+}
+
```

Comment:
IMO a new type to support the implementation of `NewMiddleware` is a bit overkill. You could implement `NewMiddleware` with closures all the way down:

```go
func NewMiddleware(mw func(http.ResponseWriter, *http.Request)) MiddlewareFunc {
	return func(handler http.Handler) http.Handler {
		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
			mw(w, r)
			handler.ServeHTTP(w, r)
		})
	}
}
```